jardin/database/__init__.py

```diff
@@ -15,42 +17,64 @@ class UnsupportedDriver(Exception): pass
+                              # connection stickiness during the lifetime of a session/task/job, etc.
 
     SUPPORTED_SCHEMES = ('postgres', 'mysql', 'sqlite', 'snowflake', 'redshift')
 
     @classmethod
     def connection(self, db_name):
-        if db_name not in self._connections:
-            self._connections[db_name] = self.build_connection(db_name)
-        return self._connections[db_name]
+        if db_name not in self._active_connections:
+            connections = self._connections.get(db_name)
+            if connections is None:
+                connections = self.build_connections(db_name)
+                self._connections[db_name] = connections
+            self._active_connections[db_name] = connections[0] if len(connections) == 1 else random.choice(connections)
+        return self._active_connections[db_name]
 
     @classmethod
-    def build_connection(self, name):
-        db = self.urls(name)
-        if db.scheme not in self.SUPPORTED_SCHEMES:
-            raise UnsupportedDriver('%s is not a supported driver' % db.scheme)
-        elif db.scheme == 'postgres' or db.scheme == 'redshift':
-            import jardin.database.drivers.pg as driver
-        elif db.scheme == 'mysql':
-            import jardin.database.drivers.mysql as driver
-        elif db.scheme == 'sqlite':
-            import jardin.database.drivers.sqlite as driver
-        elif db.scheme == 'snowflake':
-            import jardin.database.drivers.sf as driver
-
-        return driver.DatabaseConnection(db, name)
-
+    def build_connections(self, name):
+        connections = []
+        configs = self.db_configs(name)
+        for db in configs:
+            if db.scheme not in self.SUPPORTED_SCHEMES:
+                raise UnsupportedDriver('%s is not a supported driver' % db.scheme)
+            elif db.scheme == 'postgres' or db.scheme == 'redshift':
+                import jardin.database.drivers.pg as driver
+            elif db.scheme == 'mysql':
+                import jardin.database.drivers.mysql as driver
+            elif db.scheme == 'sqlite':
+                import jardin.database.drivers.sqlite as driver
+            elif db.scheme == 'snowflake':
+                import jardin.database.drivers.sf as driver
+            connections.append(driver.DatabaseConnection(db, name))
+        return connections
 
     @classmethod
-    def urls(self, name):
-        if len(self._urls) == 0:
+    def db_configs(self, name):
+        if len(self._db_configs) == 0:
             config.init()
-        for (nme, url) in config.DATABASES.items():
-            if url:
-                self._urls[nme] = DatabaseConfig(url)
-        return self._urls[name]
+            for (nme, urls) in config.DATABASES.items():
+                if not urls:
+                    continue
+                url_list = re.split(r'\s+', urls) if isinstance(urls, str) else [urls]
```

Comment:
if `urls` is already a `list`, should not it be `url_list = re.split(r'\s+', urls) if isinstance(urls, str) else urls`

Reply:
jardin supports two formats of DB config: string url and dictionary (snowflake). In the PR, we don’t deal with multi config in dictionary format - so just put it in array to process by map(). Let me make a comment in code

Reply:
Ok i see ! 
I thought we could also define a db config this way:
`"db_name" : [replica_url_1, replica_url_2, replica_url_3]`

Reply:
But your change makes sense !